homeassistant/components/litterrobot/__init__.py

```diff
@@ -24,7 +24,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     await hub.login(load_robots=True)
 
     if any(hub.litter_robots()):
-        await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+        await hass.config_entries.async_forward_entry_setups(
+            entry,
+            PLATFORMS
+            if hub.supports_button
+            else [platform for platform in PLATFORMS if platform != Platform.BUTTON],
```

Comment:
Please split this up. We try to avoid ternary Operators that span multiple lines.

Reply:
We should also prevent the vacuum platform from being loaded if there is only a Feeder-Robot.